sys/compat/linuxkpi/common/include/linux/err.h

```diff
@@ -37,30 +37,30 @@
 
 #define MAX_ERRNO	4095
 
-#define IS_ERR_VALUE(x) unlikely((x) >= (unsigned long)-MAX_ERRNO)
+#define IS_ERR_VALUE(x) unlikely((x) >= (uintptr_t)-MAX_ERRNO)
 
 static inline void *
-ERR_PTR(long error)
+ERR_PTR(intptr_t error)
 {
 	return (void *)error;
 }
 
-static inline long
+static inline intptr_t
 PTR_ERR(const void *ptr)
 {
-	return (long)ptr;
+	return (intptr_t)ptr;
 }
 
-static inline long
```

Comment:
Hm, should this and IS_ERR_OR_NULL still return `long` instead? Really they should be `bool`...

Reply:
In fact linux has them as `bool` https://github.com/torvalds/linux/blob/ffc1c20c46f74e24c3f03147688b4af6e429654a/include/linux/err.h#L34

Reply:
Done

Reply:
(upstream?... :D)

Reply:
I have a subset of these patches as a branch against FreeBSD I plan to throw into phab once I've tested it a bit more.  It booted RISC-V yesterday, but I need to test the bool change in it.